faunadb-java/src/main/java/com/faunadb/client/query/Pagination.java

```diff
@@ -0,0 +1,175 @@
+package com.faunadb.client.query;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.faunadb.client.types.Value;
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableMap;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A pagination expression.
+ * <p>
+ * <b>Example:</b>
+ * We can paginate all instances of a class, with its evets, in pages with 4 elements only.
+ * <pre>{@code
+ * client.query(
+ *    Paginate(Match(Ref("indexes/all_spells"))
+ *      .withEvents(true)
+ *      .withSize(4)
+ * );
+ * }</pre>
+ *
+ * @see Language#Paginate(Expr)
+ * @see <a href="https://faunadb.com/documentation/queries#read_functions">FaunaDB Read Functions</a>
+ */
+public final class Pagination extends Fn.Call {
```

Comment:
Yeah, Fn.Call probably shouldn't be a value, as excluding it from Value is the point. You shouldn't be able to create an ObjectV or ArrayV directly with exprs in them.

In the scala DSL, Obj and Arr convert their Expr args into Values and then stuff that back into a pre-escaped Expr type.